pandas/core/ops/array_ops.py

```diff
@@ -74,8 +74,9 @@ def masked_arith_op(x, y, op):
                 result[mask] = op(xrav[mask], yrav[mask])
 
     else:
-        assert is_scalar(y), type(y)
-        assert isinstance(x, np.ndarray), type(x)
+        if not is_scalar(y):
```

Comment:
Is this a behavior change from master? Or was this case not reached before?

Reply:
in tests.arithmetic.test_object we have a test

```
with pytest.raises(TypeError):
    index.tolist() - index
```

that raises AssertionError without this edit because `y` is a list at this point.  So shouldn't be a real behavior change, just a better error